### PR TITLE
Elasticsearch 7.10 upgrade

### PR DESCRIPTION
**Pull request by JenniWhitman**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
updates https://github.com/mitodl/open-discussions/issues/3494

#### What's this PR do?
This PR upgrades from ES 6.8 to ES 7.10 in preparation for the OpenSearch swap.

#### How should this be manually tested?
Pull the branch locally and run `docker build` and `docker up` to run the new version of Elasticsearch. 

I would then:
- run tests
- check that you can use the search API manually using postman or fetch or some tool of your choice - specifically as OCW might make requests
- make sure that pages within Open that are using the search API function (new open demo, infinite, learn, open discussions)

#### Where should the reviewer start?
Pull the branch and play with it, honestly. I can run all the tests in the world and I can try to break it, but ultimately it's best if we have many people poke it from all sides.

#### What GIF best describes this PR or how it makes you feel?
ImprobableFarflungHapuka-max-1mb

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -39,16 +39,18 @@ services:
...
+    image: docker.elastic.co/elasticsearch/elasticsearch:7.10.0
+    container_name: elastic
     environment:
       - network.host=0.0.0.0
       - http.cors.enabled=true
       - http.cors.allow-origin=*
       - http.max_content_length=10mb
       - rest.action.multi.allow_explicit_index=false
       - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+      - discovery.type=single-node
```
Comment (mbertrand) [inline docker-compose.yml:51]: I noticed that elastic shuts down without this line locally, but my guess is RC/prod isn't single-node.  Probably doesn't matter in terms of how the code works though, right?  I googled this setting and it doesn't seem to require any other changes.

**Review by mbertrand — COMMENTED** [review]
Looks like you need to add this to `course_catalog.etl.loaders_test.mock_upsert_tasks` in order to fix the failing test (no idea why it wasn't failing before):

```
index_run_content_files=mocker.patch("search.task_helpers.index_run_content_files"),
```

File: docker-compose.yml (hunk 1 above)
Comment (JenniWhitman) [reply docker-compose.yml:51]: This was suggested for local (and as you noticed it's unhappy without), but prod is multiple nodes - I'm pretty sure @Ardiea handles that elsewhere? Or do I need to create yet another yaml file?

**Comment by JenniWhitman** [comment]
> index_run_content_files=mocker.patch("search.task_helpers.index_run_content_files"),

Thank you - I knew there was a missing mock somewhere but was more confused how it ran prior

File: docker-compose.yml (hunk 1 above)
Comment (Ardiea) [reply docker-compose.yml:51]: No this won't affect anything outside of the local dev envs.

**Comment by abeglova** [comment]
I also tested out the search with 7.10 seems to work fine!

**Review by mbertrand — APPROVED** [review]
Works great! 👍

**Pull request merged**